compiler/sigmatch.nim

```diff
@@ -1650,7 +1650,7 @@ proc matchesAux(c: PContext, n, nOrig: PNode,
     if a >= formalLen-1 and formal != nil and formal.typ.isVarargsUntyped:
       incl(marker, formal.position)
       if container.isNil:
-        container = newNodeIT(nkBracket, n.sons[a].info, arrayConstr(c, n.info))
+        container = newNodeIT(nkArgList, n.sons[a].info, arrayConstr(c, n.info))
```

Comment:
I decided to switch this to `nkArgList`, because I found some code handling it, which was already doing the right thing (obviously left from the past).

I don't think this change is particularly risky for backwards compatibility, because it affects only macros having `varargs[untyped]` arguments. Since these will always be mapped to `nkArgList` (or `nkBracket` in the past), it's unlikely that there will be code, which checks the node kind of the macro param - most of the code probably just takes the `len` and iterates over the node's children. 

I didn't want to keep the `nkBracket`, because I found that it's indistinguishable from a passed-in array. A solution keeping it would have required defining yet another node flag.